kubectl-fdb/cmd/cordon.go

```diff
@@ -127,62 +127,38 @@ kubectl fdb cordon --node-selector machine=a,disk=fast -l fdb-cluster-label
 
 // cordonNode gets all process groups of this cluster that run on the given nodes and add them to the remove list
 func cordonNode(cmd *cobra.Command, kubeClient client.Client, inputClusterName string, nodes []string, namespace string, withExclusion bool, wait bool, clusterLabel string) error {
-	cmd.Printf("Start to cordon %d nodes\n", len(nodes))
+	cmd.Printf("Starting to cordon %d nodes\n", len(nodes))
 	if len(nodes) == 0 {
-		return nil
+		return errors.New("no nodes were provided for cordoning")
 	}
 
-	var errors []string
+	var totalRemoved int
+
 	for _, node := range nodes {
 		pods, err := fetchPodsOnNode(kubeClient, inputClusterName, namespace, node, clusterLabel)
 		if err != nil {
-			internalErr := fmt.Sprintf("Issue fetching Pods running on node: %s. Error: %s\n", node, err)
-			cmd.PrintErr(internalErr)
-			errors = append(errors, internalErr)
-			continue
+			return fmt.Errorf("issue fetching Pods running on node %s. Error: %s", node, err)
```

Comment:
Is this intended to end the program here if we see error in any of the nodes ? 
Lets say hypothetically we are running it for 10 knodes and we see error in the first one, the previous implementation will work on the remaining 9 but the current one will stop the execution immediately.

Reply:
yes, discussion here: https://github.com/FoundationDB/fdb-kubernetes-operator/pull/1936#discussion_r1484164809 
Basically, to me, it seemed like we needed to converge on one strategy (collate errors or terminate on first).  Thread discusses it more but I lean towards making it obvious that the error occurred on the first node in your scenario, and having the user retry since the action is idempotent

Reply:
The new approach sounds good to me, just wanted to make sure we are leaning towards the new approach 👍